snap/squashfs/stat.go

```diff
@@ -83,6 +84,18 @@ func fromRaw(raw []byte) (*stat, error) {
 		p++
 	}
 
+	if st.mode&os.ModeSymlink != 0 {
+		// the symlink *could* be from a file called "foo -> bar" to
+		// another called "baz -> quux" in which case the following
+		// would be wrong, but so be it.
+
+		idx := strings.Index(st.path, " -> ")
+		if idx < 0 {
+			return nil, errBadPath(raw)
+		}
+		st.path = st.path[:idx]
```

Comment:
We're so close to resolving the symlinks now. :)

Reply:
Random thought:

```
// package squashfs
type Readlinker {
    Readlink() (string, err)
}

func (s stat) Sys() { return *s }
func (s *stat) Readlink() {
     if st.Mode & os.ModeSymlink { return s.linksTo, nil }
     return "", errors.New("not a symlink")
}
```

Then while walking:
```
err := c.Walk(".", func(path string, info os.FileInfo, err error) error {
     ...
     if info.Mode() & os.ModeSymlink {
         if rd, _ := info.Sys().(squashfs.Readlinker); rd != nil {
           whereTo, err := rd.Readlink()
           ...
         }
    }
    ....
````

Reply:
it'd have to be something like adding `Readlink` to the `Container` interface, and have `snapdir` and `squashfs` both implement it, but yeah, something like that is doable. With the existing "have ` -> ` in a filename and watch it being hilarious" caveat.

As I said on IRC, if we want to do that I think it's followup material, not for this initial PR (which has already stagnated enough)